agent/xds/jwt_authn.go

```diff
@@ -22,58 +23,35 @@ const (
 	jwksClusterPrefix    = "jwks_cluster"
 )
 
-// This is an intermediate JWTProvider form used to associate
-// unique payload keys to providers
-type jwtAuthnProvider struct {
-	ComputedName string
-	Provider     *structs.IntentionJWTProvider
-}
-
+// makeJWTAuthFilter builds jwt filter for envoy. It limits its use to referenced provider rather than every provider.
+//
+// Eg. If you have three providers: okta, auth0 and fusionAuth and only okta is referenced in your intentions, then this
+// will create a jwt-auth filter containing just okta in the list of providers.
 func makeJWTAuthFilter(pCE map[string]*structs.JWTProviderConfigEntry, intentions structs.SimplifiedIntentions) (*envoy_http_v3.HttpFilter, error) {
 	providers := map[string]*envoy_http_jwt_authn_v3.JwtProvider{}
-	var rules []*envoy_http_jwt_authn_v3.RequirementRule
+	var jwtRequirements []*envoy_http_jwt_authn_v3.JwtRequirement
 
 	for _, intention := range intentions {
 		if intention.JWT == nil && !hasJWTconfig(intention.Permissions) {
 			continue
 		}
-		for _, jwtReq := range collectJWTAuthnProviders(intention) {
-			if _, ok := providers[jwtReq.ComputedName]; ok {
+		for _, jwtReq := range collectJWTRequirements(intention) {
```

Comment:
Should use another name for `jwtReq` since it's actually a provider, not a `JwtRequirement`.